bluesky/protocols.py

```diff
@@ -244,18 +326,32 @@ def check_value(self, value: Any) -> Union[NoReturn, None]:
 
         This method is used by simulators that check limits. If not implemented
         those simulators should assume all values are valid, but may warn.
+
+        This method may be used during a scan, so should not write to any Signals
         """
         ...
 
 
+class Hints(TypedDict, total=False):
+    fields: List[str]
+    dimensions: List[Tuple[List[str], Literal["primary"]]]
+    gridding: Literal["rectilinear", "rectilinear_nonsequential"]
+
+
 @runtime_checkable
-class Hinted(Protocol):
+class HasHints(HasName, Protocol):
     @property
-    def hints(self) -> Dict:
+    def hints(self) -> Hints:
         """A dictionary of suggestions for best-effort visualization and processing.
 
         This does not affect what data is read or saved; it is only
         a suggestion to enable automated tools to provide helpful information
         with minimal guidance from the user. See :ref:`hints`.
         """
         ...
+
+
+def check_supports(obj, protocol: Type[T]) -> T:
+    assert isinstance(obj, protocol), \
+        "%s does not implement all %s methods" % (obj, protocol.__name__)
+    return obj
```

Comment:
Why is this here? Is it because we don't like the way `isinstance` reads? Or is it for a prettier exception? This just jumps out at me as "why"?

Reply:
Good question. I've now added a docstring that hopefully explains it

    This exists so that multiple protocol checks can be run in a mypy 
    compatible way, e.g.::

        triggerable = check_supports(obj, Triggerable)
        triggerable.trigger()
        readable = check_supports(obj, Readable)
        readable.read()

The problem is that mypy won't allow multiple type narrowing with isinstance in a single function, so you couldn't do:

```python
assert isinstance(obj, Triggerable)
obj.trigger()
assert isinstance(obj, Readable)
obj.read()
```

The call to read() will fail type checking

Reply:
Wow, I never knew that.